sympy/core/tests/test_power.py

```diff
@@ -661,3 +662,9 @@ def test_issue_26546():
     assert Pow(x+I, Rational(1,2)).is_extended_real is False
     assert Pow(x+I, Rational(1,13)).is_extended_real is False
     assert Pow(x+I, Rational(2,3)).is_extended_real is None
+
+
+def test_issue_25165():
+    e1 = (1/sqrt(( - x + 1)**2 + (x - 0.23)**4)).series(x, 0, 2)
+    e2 = 0.998603724830355 + 1.02004923189934*x + O(x**2)
```

Comment:
will e2 have floats if the change of floats to rational in g takes place?

Reply:
well, for this example the it never enters the conditional as there the no issue with cancellation in `(b-f)`, and leadterm calculation takes place fine to get a non-zero d.